### PR TITLE
Refactor Service/Agency dropdown list to use new `Agency` table

### PR DESCRIPTION
**Pull request by juliusfitzhugh-ccpo**
Refactor dropdown list for selecting an agency to use new `Agency` table instead of choices from `Organization.service_agency` column.

### Main refactor
- Add Agency table api
- Replace choice records with agency records in dropdown selection
  - Selecting an agency from the dropdown works
  - Typing out an agency name also works

### Updating names
- Update store file to use new naming related to agency
- Update portfolio components to use new naming for agency
- Update steps 1 and 6 components to use new naming for agency
- Update dashboard and service to use new naming for agency
- Update cypress to use new naming for agency

The main name changes that occurred in the files to reflect the use of the `Agency` table is shown below
- `selectedServiceOrAgency` -> `selectedAgency`
- `serviceAgency` -> `agency`
- `serviceOrAgency` -> `agency`
- `Service/Agency` -> `Agency`
- `service_agency_data` -> `agency_data`
- `service_agency` -> `agency`

### Unit tests
- Add/update unit tests for most of file that had changes
- Initial issue with the `JWCC` component when trying to run unit test
  - Error: `Canvas is already in use. Chart with ID '0' must be destroyed before the canvas can be reused.`
  - Fix: change `mount()` in the `BarChart` component into `async/await` when creating the chart. 
- the `JWCC` component unit test is only an initial start to unit testing

Ticket: AT-7965

### CODE REVIEW

File: src/store/portfolio/index.ts [hunk 1]
```diff
@@ -143,7 +143,7 @@ export class PortfolioDataStore extends VuexModule {
         phoneNumber:"5555555555",
         phoneExt:"1234",
         designation: "Civilian",
-        serviceAgency: "U.S Army"
+        agency: "U.S. Army"
```
Comment (jasonburkert-ccpo) [inline src/store/portfolio/index.ts:146]: Nice catch

File: src/dashboards/JWCC.vue [hunk 2]
```diff
@@ -412,7 +412,7 @@ export default class JWCCDashboard extends Vue {
   public getCurrencyString = getCurrencyString;
 
   public async calculateSpendRateLineChartData(): Promise<void> {
-    const agencies = this.fundsSpentByServiceAgency.map(a => a.name);
+    const agencies = this.fundsSpentByAgency.map(a => a.name);
```
Comment (tonysumner-ccpo) [inline src/dashboards/JWCC.vue:415]: In the screencap below, something is preventing name from being populated on the JWCC page.  I believe once that is resolved, there will be data for the charts on the JWCC page.

To get to the JWCC page, look at the 2nd screencap

console error on JWCC page
image

How to access the JWCC page
image

**Comment by tonysumner-ccpo** [comment]
One more thing, please alphabetize the items by `label` before they are retrieved and sent to the front end.

File: src/dashboards/JWCC.vue (hunk 2 above)
Comment (juliusfitzhugh-ccpo) [reply src/dashboards/JWCC.vue:415]: Fixed in https://github.com/dod-ccpo/atat-web-ui/pull/1055/commits/498704fdd2ff23bf95c606c326c551c50b5ea0a1 see comment for more details. In summary,
- old `Service/Agency` still lingering in `develop`, updated
- update request to get the `agency.title` for each agency

**Review by ericyoungquist-ccpo — APPROVED** [review]

**Pull request merged**